### PR TITLE
Add package to convert number of milliseconds to a duration string

### PR DESCRIPTION
**Pull request by Planeshifter**
Resolves # .

## Description

> What is the purpose of this pull request?

This pull request:

-   adds a package to convert number of milliseconds a to duration string

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #
-   fixes #

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by Planeshifter** [comment]
This package is now ready for review.

File: lib/node_modules/@stdlib/time/ms2duration/README.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+-   A duration string is a string containing a sequence of time units. A time unit is a non-negative integer followed by a unit identifier. The following unit identifiers are supported:
+
+    -   `d`: days
+    -   `h`: hours
+    -   `m`: minutes
+    -   `s`: seconds
+    -   `ms`: milliseconds
+    
+    For example, the string `1m3s10ms` is a duration string containing three time units: `1m` (1 minute), `3s` (3 seconds), and `10ms` (10 milliseconds). The string `60m` is a duration string containing a single time unit: `60m` (60 minutes). A duration string must contain at least one time unit.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/time/ms2duration/README.md]:
```suggestion
    For example, the string `1m3s10ms` is a duration string containing three time units: `1m` (1 minute), `3s` (3 seconds), and `10ms` (10 milliseconds). The string `60m` is a duration string containing a single time unit: `60m` (60 minutes).
```

File: lib/node_modules/@stdlib/time/ms2duration/lib/main.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	var out;
+	if ( !isNonNegativeInteger( ms ) ) {
+		throw new TypeError( 'invalid argument. Must provide a nonnegative integer. Value: `' + ms + '`.' );
+	}
+	if ( ms === 0 ) {
+		return '0ms';
+	}
+	out = '';
+	if ( ms >= 86400000 ) {
+		out += (ms/86400000|0) + 'd';
```
Comment (kgryte) [inline lib/node_modules/@stdlib/time/ms2duration/lib/main.js]: We should use `floor` here. This only works is the result of division fits within a 32-bit int. Should also consider what to do if number serializes in exponential notation.

In short, values should always round-trip. Based on what I know of the inverse implementation, handling exponential notation is not supported (nor should it be).

File: lib/node_modules/@stdlib/time/ms2duration/lib/main.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+		ms %= 86400000;
+	}
+	if ( ms >= 3600000 ) {
+		out += (ms/3600000|0) + 'h';
```
Comment (kgryte) [inline lib/node_modules/@stdlib/time/ms2duration/lib/main.js]: This should be fine; however, one has to remember operator precedence. My preference would be to use floor for this and similar operations, as it is more explicit in terms of what is happening.

**Review by kgryte — CHANGES_REQUESTED** [review]

File: lib/node_modules/@stdlib/time/ms2duration/lib/main.js (hunk 2 above)
Comment (Planeshifter) [reply lib/node_modules/@stdlib/time/ms2duration/lib/main.js]: Changed to use `floor`.

File: lib/node_modules/@stdlib/time/ms2duration/lib/main.js (hunk 3 above)
Comment (Planeshifter) [reply lib/node_modules/@stdlib/time/ms2duration/lib/main.js]: Changed to use `floor`.

File: lib/node_modules/@stdlib/time/ms2duration/lib/main.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+* // returns '1s30ms'
+*
+* @example
+* var str = ms2duration( 3600000 );
+* // returns '1h'
+*/
+function ms2duration( ms ) {
+	var out;
+	if ( !isNonNegativeInteger( ms ) ) {
+		throw new TypeError( 'invalid argument. Must provide a nonnegative integer. Value: `' + ms + '`.' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/time/ms2duration/lib/main.js]: @Planeshifter Missing `format`.

**Review by kgryte — APPROVED** [review]
LGTM. Will merge after all CI jobs pass.

**Pull request merged**